File: trunk/EngineUAB/Run/Videogame/Data/Scripts/Game/Player/player_animation_jump_state.lua
class 'CPlayerAnimationJumpState' (CState)
	function CPlayerAnimationJumpState:__init() 
		CState.__init(self)
	end

	function CPlayerAnimationJumpState:OnEnter(_CCharacter)
		if not ( _CCharacter == nil ) then
			num = _CCharacter:get_animation_id("jump")
			_CCharacter:get_animation_model():execute_action( num, 1.0 )
		end
	end
	
	function CPlayerAnimationJumpState:Execute(_CCharacter)
	end
	
	function CPlayerAnimationJumpState:OnExit(_CCharacter)
	end
	
	function CPlayerAnimationJumpState:OnMessage(_CCharacter, _Msg)
		return false
	end
	
	function CPlayerAnimationJumpState:__Finalize()
	end

File: trunk/EngineUAB/Code/Videogame/Characters/Player/AnimationStates/PlayerAnimationJumpState.cpp
#include "PlayerAnimationJumpState.h"
#include "StatesMachine\Telegram.h"
#include "RenderableObjects\AnimatedModel\AnimatedInstanceModel.h"

#if defined _DEBUG
#include "Memory\MemLeaks.h"
#endif

CPlayerAnimationJumpState::CPlayerAnimationJumpState( const std::string &_Name )
	: CState(_Name)
{

}

CPlayerAnimationJumpState::~CPlayerAnimationJumpState()
{

}

void CPlayerAnimationJumpState::OnEnter( CCharacter* _pCharacter )
{
	int l_iAnimID = _pCharacter->GetAnimationID("jump");
	_pCharacter->GetAnimatedModel()->ExecuteAction(l_iAnimID, 1.0f);
}

void CPlayerAnimationJumpState::Execute( CCharacter* _pCharacter, float _fElapsedTime )
{

}

void CPlayerAnimationJumpState::OnExit( CCharacter* _pCharacter )
{

}

bool CPlayerAnimationJumpState::OnMessage( CCharacter* _pCharacter, const STelegram& _Message )
{
	return false;
}